src/ophyd_async/plan_stubs/_fly.py

```diff
@@ -126,18 +125,8 @@ def fly_and_collect(
 
     done = False
     while not done:
-        try:
-            yield from bps.wait(group=group, timeout=0.5)
-        except TimeoutError:
-            pass
-        else:
-            done = True
-        yield from bps.collect(
-            *detectors,
-            return_payload=False,
-            name=stream_name,
-        )
-    yield from bps.wait(group=group)
+        done = yield from bps.wait(group=group, timeout=flush_period, move_on=True)
+        yield from bps.collect(*detectors, name=stream_name)
```

Comment:
Is the removal of `return_payload` intentional?

Reply:
I have checked this today...This is working ... I'm able to collect data

Reply:
Can't see the full context for this since the github diff is cluttered with all the changes to main since this PR was opened. If you open a new PR after rebasing then add me as a reviewer :pray:

Reply:
I have created a new PR https://github.com/bluesky/ophyd-async/pull/691 because the branch go messed up...